### PR TITLE
Bugfix/font stubs

### PR DESCRIPTION
**Pull request by dr0id**
follow up PR of #2175 to fix the stubs of all properties in font and freetype

### CODE REVIEW

**Comment by dr0id** [comment]
> What am I looking at?
> 
> I see a bazillion formatting changes, and attributes are `@property` not in the font stubs. Why?

The changes are in 
gen_stubs.py   (to be consistent with formatting)
font.pyi
freetype.pyi


All the other changes are just formatting changes after running:
setup.py lint
setup.py format

Not sure why I have those changes, but ci builds seems not to bothered by those changes. 

I'm using:
Python 3.10.4
black                         23.7.0
clang                         16.0.1.1
clang-format                  16.0.4

**Comment by ankith26** [comment]
RN `setup.py lint` doesn't really format the typestubs, perhaps your formatting is a result of something else? Anyways, for ease of review I'd like the formatting changes in a separate PR (I know... It's ironic coming from me :smile:)

**Comment by dr0id** [comment]
> RN `setup.py lint/format` doesn't really format the typestubs, perhaps your formatting is a result of something else? Anyways, for ease of review I'd like the formatting changes in a separate PR (I know... It's ironic coming from me 😄)

You are right, I run (from my setup.py):

`python.exe -m black buildconfig\stubs\pygame`

Right now from the setup.py only following python directories would be touched: 

`        python_directories = ["src_py", "test", "examples"]`

**Comment by dr0id** [comment]
> RN `setup.py lint/format` doesn't really format the typestubs, perhaps your formatting is a result of something else? Anyways, for ease of review I'd like the formatting changes in a separate PR (I know... It's ironic coming from me 😄)

removed linting changes @ankith26

File: buildconfig/stubs/pygame/freetype.pyi [hunk 1]
```diff
@@ -38,34 +38,98 @@ STYLE_WIDE: int
...
-    origin: bool
-    pad: bool
-    ucs4: bool
-    resolution: int
+    @property
+    def size(self) -> Union[float, Tuple[float, float]]: ...
+    @size.setter
+    def size(self, value: Union[float, Tuple[float, float]]) -> None: ...
+    @property
+    def style(self) -> None: ...
```
Comment (ankith26) [inline buildconfig/stubs/pygame/freetype.pyi:46]:
```suggestion
    def style(self) -> int: ...
```

File: buildconfig/stubs/pygame/freetype.pyi [hunk 2]
```diff
@@ -38,34 +38,98 @@ STYLE_WIDE: int
...
+    @rotation.setter
+    def rotation(self, value: int) -> None: ...
+    @property
+    def fgcolor(self) -> Color: ...
+    @fgcolor.setter
+    def fgcolor(self, value: Color) -> None: ...
+    @property
+    def bgcolor(self) -> Color: ...
+    @bgcolor.setter
+    def bgcolor(self, value: Color) -> None: ...
```
Comment (ankith26) [inline buildconfig/stubs/pygame/freetype.pyi:128]: The setter for this (and above `fgcolor`) should be `ColorValue`

File: buildconfig/stubs/pygame/freetype.pyi (hunk 1 above)
Comment (dr0id) [reply buildconfig/stubs/pygame/freetype.pyi:46]: thanks for spotting that

File: buildconfig/stubs/pygame/freetype.pyi (hunk 2 above)
Comment (dr0id) [inline buildconfig/stubs/pygame/freetype.pyi:128]:
```suggestion
    @property
    def fgcolor(self) -> ColorValue: ...
    @fgcolor.setter
    def fgcolor(self, value: ColorValue) -> None: ...
    @property
    def bgcolor(self) -> ColorValue: ...
    @bgcolor.setter
    def bgcolor(self, value: ColorValue) -> None: ...
```

Thanks for spotting this.

File: buildconfig/stubs/pygame/freetype.pyi [hunk 3]
```diff
@@ -38,34 +38,98 @@ STYLE_WIDE: int
...
+    @property
+    def rotation(self) -> int: ...
+    @rotation.setter
+    def rotation(self, value: int) -> None: ...
+    @property
+    def fgcolor(self) -> ColorValue: ...
+    @fgcolor.setter
+    def fgcolor(self, value: ColorValue) -> None: ...
+    @property
+    def bgcolor(self) -> ColorValue: ...
```
Comment (ankith26) [inline buildconfig/stubs/pygame/freetype.pyi:126]: I only meant for the setters to take `ColorValue`. Getters should still return `Color` (a `ColorValue` is a broader type that is the union of types that could be handled as Colors, but `Color` is the exact type)

Comment (dr0id) [inline buildconfig/stubs/pygame/freetype.pyi:126]:
```suggestion
    def fgcolor(self) -> Color: ...
    @fgcolor.setter
    def fgcolor(self, value: ColorValue) -> None: ...
    @property
    def bgcolor(self) -> Color: ...
```

Comment (dr0id) [reply buildconfig/stubs/pygame/freetype.pyi]: Ah, that makes sense. Thanks for the explanation.

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :tada:

**Review by MyreMylar — APPROVED** [review]
LGTM 👍 

Clear what the change is now, and I approve.

**Pull request merged**